backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/SnapshotsManager.java

```diff
@@ -344,16 +345,16 @@ private void populateDisksWithVmData(List<? extends Disk> disks, Guid vmId) {
         }
     }
 
-    private void addVmExternalData(Map<VmExternalDataKind, String> vmExternalData, VM vm) {
+    private void addVmExternalData(Map<VmExternalDataKind, SecretValue<String>> vmExternalData, VM vm) {
         if (VmDeviceCommonUtils.isVmDeviceExists(vm.getStaticData().getManagedDeviceMap(), VmDeviceType.TPM)) {
-            String tpmData = vmDao.getTpmData(vm.getId()).getFirst();
-            if (tpmData != null && !tpmData.equals("")) {
+            SecretValue<String> tpmData = vmDao.getTpmData(vm.getId()).getFirst();
+            if (tpmData != null && !tpmData.getValue().equals("")) {
```

Comment:
`SecretValue.isEmpty(tpmData)` would be useful here.

Reply:
But, looking into `VmDaoImpl.getTpmData()` I don't see any case when it may return `null`. But `tpmData.getValue()` may be `null`.